test/typescript/dispatch.ts

```diff
@@ -1,12 +1,15 @@
 import {Dispatch, Action} from "../../index.d.ts";
 
 
-declare const dispatch: Dispatch;
-
+declare const dispatch: Dispatch<any>;
 
 const dispatchResult: Action = dispatch({type: 'TYPE'});
 
+// thunk
+declare module "../../index.d.ts" {
+    export interface Dispatch<S> {
```

Comment:
The `S` here is a part of dispatch's contract defined in the `index.d.ts`. I've added it because some results may depend on it. And, moreover, `S` is always known by dispatch - its a part of typed store.